src/wasm.h

```diff
@@ -90,6 +90,9 @@ struct Name : public cashew::IString {
 
 // An index in a wasm module
 typedef uint32_t Index;
+inline Index toIndex(size_t index) {
```

Comment:
Yes, let's add an assert here. But I think its enough to check on the maximum size of the type (a more specific check would require some overhead in getting that information to here).